### PR TITLE
Feature print(send) stop reason

### PR DESCRIPTION
**Pull request by normanlinnet**

### CODE REVIEW

File: internal/service/character/fr_arbitrage.go [hunk 1]
```diff
@@ -180,8 +180,12 @@ func (fra *FRArb) genSignal(future *future) (bool, bool) {
 	shouldStop = future.size != 0 && ((future.size*nextFundingRate > 0 && -nextFundingAPR <= fra.stopAPRThreshold) ||
 		outerSpreadRate <= fra.stopFutureSpotSpreadRate)
 	if shouldStop {
-		util.Info(fra.tag, "not profitable: "+future.name)
-		fra.send("not profitable: " + future.name)
+		stopReason := "future.size * nextFundingRate > 0 && -nextFundingAPR <= fra.stopAPRThreshold"
+		if outerSpreadRate <= fra.stopFutureSpotSpreadRate {
+			stopReason = "outerSpreadRate <= fra.stopFutureSpotSpreadRate" // please update string when shouldStop logic change
+		}
```
Comment (Kobe-Wang) [inline internal/service/character/fr_arbitrage.go:186]:
```go
message := fmt.Sprintf("not profitable: %s \n stop reason: %s", future.name, stopReason)
util.Info(fra.tag, message)
fra.send(message)
```

Comment (normanlinnet) [reply internal/service/character/fr_arbitrage.go:186]: 9d6c4dcd0008cf8f56b7bdc3c410a4fb5eaf1a50 Thanks

File: internal/service/character/fr_arbitrage.go [hunk 2]
```diff
@@ -180,8 +180,13 @@ func (fra *FRArb) genSignal(future *future) (bool, bool) {
 	shouldStop = future.size != 0 && ((future.size*nextFundingRate > 0 && -nextFundingAPR <= fra.stopAPRThreshold) ||
 		outerSpreadRate <= fra.stopFutureSpotSpreadRate)
 	if shouldStop {
-		util.Info(fra.tag, "not profitable: "+future.name)
-		fra.send("not profitable: " + future.name)
+		stopReason := "future.size * nextFundingRate > 0 && -nextFundingAPR <= fra.stopAPRThreshold"
```
Comment (CheshireCatNick) [inline internal/service/character/fr_arbitrage.go:183]: "funding rate not profitable"

File: internal/service/character/fr_arbitrage.go [hunk 3]
```diff
@@ -180,8 +180,13 @@ func (fra *FRArb) genSignal(future *future) (bool, bool) {
...
+		if outerSpreadRate <= fra.stopFutureSpotSpreadRate {
+			stopReason = "outerSpreadRate <= fra.stopFutureSpotSpreadRate" // please update string when shouldStop logic change
```
Comment (CheshireCatNick) [inline internal/service/character/fr_arbitrage.go:185]: "outer spread smaller than threshold"

**Pull request merged**